src/topology/mod.rs

```diff
@@ -127,23 +146,33 @@ impl RunningTopology {
         .map_err(|_| ())
     }
 
-    pub fn reload_config(
+    pub fn load_config_and_spawn(
         &mut self,
-        new_config: Result<Config, Vec<String>>,
+        new_config: Config,
         rt: &mut tokio::runtime::Runtime,
+        exit_after: Option<Stage>,
         require_healthy: bool,
-        initial_load: bool,
     ) -> bool {
-        let new_config = match new_config {
-            Ok(config) => config,
-            Err(errors) => {
-                for error in errors {
-                    error!("Configuration error: {}", error);
-                }
-                return false;
-            }
-        };
+        self.reload_config_and_respawn(new_config, rt, exit_after, require_healthy, true)
+    }
 
+    pub fn reload_config_on_hot(
```

Comment:
not sure this fn name makes much sense to me, could you explain what this is supposed to do? I see it just calls out to another fn, why not use that one?

Reply:
Actually, I haven't noticed that after some refactoring new method `load_config_and_spawn` became unused. It is a symmetrical method to `reload_config_on_hot`. Both of them appeared as splitting `reload_config` into cases for initial load and actual reloading.

I'm removing `load_config_and_spawn` now. But I would like to leave `reload_config_on_hot` here, because it hides unnecessary details from user of `topology` module: when we really reloading config we already don't care about `exit_after` and `initial_load` parameters, they are always `None` and `false`. This makes usage of this functionality more ergonomic (we have 14 usages of this method in tests, btw).

I agree that the name of the method can be confusing, I couldn't invent better one... We can rename it.

Reply:
Can we rename it to `reload_config_and_respawn`?

Reply:
Yep, renaming this into `reload_config_and_respawn`. And current `reload_config_and_respawn` into `load_config_and_respawn` since it is used both for reloading and initial loading.